src/Misc/layoutbin/en-US/strings.json

```diff
@@ -559,6 +559,7 @@
   "SessionExistStopRetry": "Stop retry on SessionConflictException after retried for {0} seconds.",
   "SetBuildVars": "Set build variables.",
   "SetEnvVar": "Setting environment variable {0}",
+  "SetVariableNotAllowed": "Setting the variable '{0}' is not allowed.",
```

Comment:
Might be nice to add a little extra detail on why it isn't allowed. Something like `Setting variable '{0}' has been disabled by the task definition or build definition`